src/Data/Auth/Providers/AuthContext.ts

```diff
@@ -0,0 +1,51 @@
+import { createContext } from "react";
+
+/**
+ * Interface for the authentication context properties.
+ */
```

Comment:
provider, not context. We need to stay clear because the two terms start to mingle in my head. 
The Context returns a CreateContext that consumes a Provider. 

This means that this file should look like so (minus the comments) instead:
```ts
import { createContext } from "react";

// or AuthProviderInterface, whichever is fine for me.
interface GetAuthProvider {
  getUser: () => void;
  login: () => void;
  logout: () => void;
  updateUser: (user: string, token: string) => void;
  getToken: () => string | null;
}

export const GetAuthContext = createContext<GetAuthProvider>({
  getUser: () => undefined,
  login: () => {},
  logout: () => {},
  updateUser: (_user: string, _token: string) => {},
  getToken: () => null, // or the return "no token"; if you consider the default being without authentication.  although I'm not certain that would be needed, or make sense to return "no token" instead of null
});

```